Objects/bytesobject.c

```diff
@@ -1867,16 +1867,19 @@ bytes_join(PyBytesObject *self, PyObject *iterable_of_bytes)
 }
 
 PyObject *
-PyBytes_Join(PyObject *sep, PyObject *x)
+PyBytes_Join(PyObject *sep, PyObject *iterable)
 {
     if (sep != NULL && !PyBytes_Check(sep)) {
         PyErr_Format(PyExc_TypeError,
                      "sep: expected bytes, got %T", sep);
         return NULL;
     }
-    assert(x != NULL);
+    if (iterable == NULL) {
+        PyErr_BadInternalCall();
+        return NULL;
+    }
```

Comment:
What if just remove this? Would not `stringlib_bytes_join` raise the same exception anyway?

Reply:
Right it works, I removed it.